Full notation IPv6 addresses

xschily opened the pull request:
Tags: fix, dev, Github issue #7

Change-Id: Ib87badcbd14c6607d2119557f2adc0afa6e928f8
Signed-off-by: Ilya Schepin ischepin@mera.ru

In Quagga function `strcmp` is used to compare both ipv4 and ipv6 addresses. This is incorrect, since there are multiple string representations of one ipv6 address (e.g. strings **2002::a1**, **2002:0:0:0:0:0:0:a1**, **2002::00a1**, **2002::00A1** represent one ipv6 address, **2002:0000:0000:0000:0000:0000:0000:00a1**.  Possible representations are described in [RFC 5952)](https://tools.ietf.org/html/rfc5952). To correctly compare ipv6 addresses, they must be compared using `IPV6_ADDR_CMP` function (`defined in lib/prefix.h`), which will compare addresses stored in `struct in6_addr` form.  
In some cases calling function is not aware of underlying string address family (**AF_INET** or **AF_INET6**). So wrapping function `ip_addr_cmp` is created, to determine AF of IP address string.

sirigadi committed:
Updating IPv6 nexthop in BGP nexthop table for 'network X:X:X:X" command

Change-Id: I55d9fdd950412f314de7bd4eb7e17f203976dac7
Signed-off-by: sirigadi <maheshwari.sirigadi@hpe.com>

duttanir committed:
new: dev: ZEBRA VRF static routes with nexthop ip

Change-Id: I5ba8aaacf13d156522572297dd75272320406216
Signed-off-by: Nirmalya Dutta <nirmalya.dutta@hpe.com>

sirigadi committed:
BGP Quagga changes for commands match ipv6 address prefix-list, match
community and match extcommunity

Change-Id: I11220818bd605ad36d14e023f5856b2b8971f3c5
Signed-off-by: sirigadi <maheshwari.sirigadi@hpe.com>

vijavaru committed:
Quagga Support for BGP Prefix List, Community Filter
lists and Redistribute

Change-Id: Ibebd0318e76ac4f8268add732afd3cc87e89ad4b
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

Commit:
new: dev: OSPFv2 Changes

Change-Id: I1d6d260d5cf829b7f866cfac1d910c8c59f1dcfe
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

Commit:
new: usr: Neighbor prefix-list support for Quagga BGP

Change-Id: I3fbb2eba5347f2cf4dd0bee9c2063f1b62ec2022
Signed-off-by: kiranvaddey <venkata.sai.vaddey@hpe.com>

sirigadi committed:
Fix for BGP CT script failure "ops-quagga.ops.tests.test_bgp_ct_maximum_paths"

Change-Id: I217e0330143082c8ef492b3a7fa7bb56fb09d57a
Signed-off-by: sirigadi <maheshwari.sirigadi@hpe.com>

vijavaru committed:
Removing Commmunity Filter Quagga Support to make
changes according to the new schema changes for the community filter

Change-Id: Ia5309f2601531252d433b9b440663d433979e517
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

Commit:
new: usr: ops-quagga support for ip aspath filters, nbr filter-list

This review has Quagga support implementation for 'ip as-path access-list' and
'neighbor filter-list'

Change-Id: I500f56d7395824be96987d488b16fc4c535f697c
Signed-off-by: kiranvaddey <venkata.sai.vaddey@hpe.com>

iyergan committed:
Quagga changes for
routemap "set extcommunity rt" and
"set extcommunity soo" CLI's

Change-Id: I3edb4ea6a1c5d2a2019839a71065e6d7971f490e
Signed-off-by: iyergan <ganesh.m.iyer@hpe.com>

vijavaru committed:
Quagga Support for BGP Community Filter CLI

Change-Id: Ifb0f7d56d3d3c51dd32080935a8e488a7d6fb75b
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

Commit:
fix: dev: Fix crash when network info is NULL

Change-Id: I23686ee349292d3488c40b1c789ed4918bbbaff9
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

Commit:
new: dev: OSPFv2 Changes (Part 2)

Incorporated following changes:
1. Adding route to RIB
2. OSPF intervals configuration
3. Enable OSPF route handling in Zebra

Change-Id: I74998a52c71b806d0587d70ac41504c92f43ed23
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

memon4hp committed:
new: dev: OSPFv2 rib OVSDB update

OSPFv2 daemon side changes for the OSPF_Route table update

Change-Id: I261ac2dc94378c9e6eafe5fd29c39808205825ae
Signed-off-by: Mahammadnaeem Memon <memon.mk@hpe.com>

HemaliWaghode committed:
new: dev: BGP Quagga changes
commands: set as-path exclude/prepend, set ipv6 next-hop global,
set origin, match as-path,match origin,match metric, match ipv6 next-hop,
match probability

Change-Id: I4aff73c02bc2ec1629b3b5de95a564a4dbf63467
Signed-off-by: waghodeh <hemali.waghode@hpe.com>

memon4hp committed:
chg: dev: OSPFv2 - optimization on incremental upd

Optimize the OVSDB updates on incremental updates of the external
routes learning.
Correction in the output of the "show ip ospf route" command for
router ids without prefixlen.
Aligned the earlier patch(es) with the coding guidelines.

Change-Id: If58c210f12b4aab327c0107f26d5f2201a74c811
Signed-off-by: Mahammadnaeem Memon <memon.mk@hpe.com>

Commit:
new: dev: OSPFv2 Changes (Part 3)

Incorporated following changes:
1. Area level and interface level authentication support.
2. Plain text and multikey MD5 key configuration support.
3. Stubby, totally stubby and NSSA area type support.
4. OSPF Administrative distance configuration for inter-area, intra-aera and external routes.

Change-Id: Ied38fe91a33711cab795efbf1a4bf75a98bc5594
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

HemaliWaghode committed:
new: dev: BGP CLI quagga changes

commands added: set atomic-aggregate, set aggregator as,
set comm-list delete, set community, set weight, set local-preference

Change-Id: I8bcbd11aa2074c272e7daab767c45cc016184f33
Signed-off-by: waghodeh <hemali.waghode@hpe.com>

vijavaru committed:
fix : dev: BGP Bug Fix

- Fixing the following defects
  - AGM 592,Taiga 315
  - Taiga 394
  - AGM 667,Taiga 420

Change-Id: I8860545a1f350fdd669570fdb46d18f62fe919d5
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

iyergan committed:
new: usr: Quagga changes for BGP clear commands
This commit includes Quagga changes for BGP commands below:
clear bgp (A.B.C.D|X:X::X:X) soft in
clear bgp (A.B.C.D|X:X::X:X) soft out
clear bgp * soft in
clear bgp * soft out
clear bgp <1-4294967295> soft in
clear bgp <1-4294967295> soft out
clear bgp peer-group WORD soft in
clear bgp peer-group WORD soft out
clear bgp ipv6 (A.B.C.D|X:X::X:X) soft in
clear bgp ipv6 (A.B.C.D|X:X::X:X) soft out
clear bgp ipv6 peer-group WORD soft in
clear bgp ipv6 peer-group WORD soft out

Change-Id: I0e65dd81acb02eb222de155425fa2b225ad10afa
Signed-off-by: iyergan <ganesh.m.iyer@hpe.com>

Nimisha-Mathews committed:
fix: usr: ALM Bug ID 1536
Routes converging and session disconnecting

When large number of routes have to be learnt by a DUT, the routes learnt
increment, reach a maximum, and the session disconnects. This fix resolves
the issue, and the session remains Established.

Change-Id: I108037699cd3e3155987adb1f79718bcba60ca7a
Signed-off-by: Nimisha Sara Mathews <nimisha.sara.mathews@hpe.com>

gauravgupta4 committed:
fix: usr: Tagia Bug#609 Zebra adding a next hop when interface is down

Zebra backend does not call the OVSDB route and next-hop selected bit set
function in all cases when zebra is trying to modify the kernel with the
route and its next-hops. In the fix, we are trying to call the route and
next-hop update function in all scenarios.
The CT script also adds test cases as reported in the bug.

Change-Id: Ib274165c23d45709136fb9116f9c47deb9f8f94b
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

vijavaru committed:
chg: dev : Support for updating the bgp router_id from ops-quagga
into OVSDB

Change-Id: I5de470726a2e988471eca9bb6e24d31aa3e052c1
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

jeremybbrown committed:
Eliminate build race condition for libzebra_ovsdb_if.a

Add a dependency to ensure librariy is built before libzebra

Change-Id: I01d8a41c6768aad12f7c70fc742eb1f3d0649d96

Commit:
new: dev: OSPFv2 route redistribution support

Change-Id: I3c9ba4876934acbd2924c15860b3ae852212b1dc
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

vijavaru committed:
fix: dev: Fix for Taiga defect 717

Change-Id: I9ce6af67afc8bf7a9762ba5adc48fc7f6002dba8
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

paulamo committed:
new: dev: Event logging for OSPF

Adding event logging for various ospf events

Change-Id: I797deaed8b98065e7f23a29decfcac7561062e4c
Signed-off-by: paulamo <amos.j.paul@hpe.com>

singhupe committed:
new: dev: OSPFv2 debug support.

Added OSPFv2 debugging support for various sub-modules like,
packet, event, ISM, NSM etc.

Change-Id: I258628310b67310da071e970dbb571284966fc42
Signed-off-by: Upendra Singh <upendra.singh@hpe.com>

gauravgupta4 committed:
fix: usr: Tagia Bug#610 ops-zebra crash when converting interface from L3 to L2

As a part of the fix, we moved all the local cached pointer based lookup into
the idl port row and route row to uuid based lookup. This conversion helped
in fixing the zebra crash.

Change-Id: Iac10bcf4161609b37407ba84f93e0aa832ab5143
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

memon4hp committed:
chg: dev: OSPFv2 removal of dependency on zebra

OSPFv2 -removal of dependency on zebra for
Port add/delete
Port status change up/down
Port ip address add/delete

Change-Id: I2f1ab43c06456af65976e676d226889c3f38c911
Signed-off-by: Mahammadnaeem Memon <memon.mk@hpe.com>

mohamedfahim committed:
New:Dev:ospfv2 FT testcases

Change-Id: Ia3dd6791653ce47fb85dd2dc2a4f779d7954e873
Signed-off-by: Mohamed Fahim <mohamed.fahim@hpe.com>

Commit:
new: dev: OSPFv2 Virtual link support

Change-Id: I8970eb9fa9d2020029e9bae854302cf66ed95820
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

gauravgupta4 committed:
new: dev: ops-zebra restartability changes

-Added support for retaining zebra routes in case zebra
 receives a shudown signal.
-Cleaned-up code in netlink interface for ops-zebra to
 correctly read the netlink route get messages to store
 zebra programmed routes in a route table
-Added upport for comparing the zebra route learned from
 kernel and routes learned from OVSDB. The comparison
 result allows us to clean-up stale routes from kernel and
 not program zebra routes which are already present in kernel.

Change-Id: Ida050a5972106b92d903d8b4cc1b9d494711c072
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

HemaliWaghode committed:
chg: dev: BGP performace

Integrate “write” side of the sockets between OVS poll loop
and Quagga “thread”

Change-Id: I47b48ca7f1b35e39aece1a4eeecad5c3ba1a2c27
Signed-off-by: waghodeh <hemali.waghode@hpe.com>

mchpe committed:
new: dev: Corresponding daemon changes for below OSPF router command.

default-metric
lsa-timer-inetrval
set area stub default-cost

Change-Id: Iaf31d00d3a1b3ee624ebcbd32ad2185918fbd045
Signed-off-by: Dinesh MC <dinesh.m-c@hpe.com>

ashwinramani committed:
new: dev: Adding supportability for zebra static route, port and nexthop

Change-Id: I19b9b31198c240089a9368ea88735b90ffca06a1
Signed-off-by: Ashwin Ramani <ashwin.ramani@hpe.com>

paulamo committed:
chg: dev:Adding custom validation for ospf

Adding restd custom validators for ospf

Change-Id: If3521c9373d2bc42ea3af9b45fed5c9e24042a00
Signed-off-by: paulamo <amos.j.paul@hpe.com>

suganya-jsa committed:
new:dev: OSPFv2 interface command daemon changes

Change-Id: I165cd30ea60e28f810eebe941ed2015069faa0a5
Signed-off-by: Suganya J S A <suganya.j-s-a@hpe.com>

gauravgupta4 committed:
chg: dev: removing redundant lines of code from zebra

Due to some merge errors while patching the diff, three
lines of redundant code got merged into master. This commit
is just removing those redundant lines.

Change-Id: I8da208389394df466448d6c8a21d5b34e48307a3
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

mohamedfahim committed:
New:Dev:ospfv2 FT testcases

Change-Id: I4ac830d4543edb6e9cb78ea602abb51b0e464057
Signed-off-by: Mohamed Fahim <mohamed.fahim@hpe.com>
Signed-off-by: Suganya J S A <suganya.j-s-a@hpe.com>

gauravgupta4 committed:
fix: usr: Tagia Bug#1016 ops-zebra crash while dereferencing null pointer

While reading the OVSDB route table ops-zebra sometimes crashes if the
next-hops for a given route are pointing to null. ops-zebra attempts to
dereferece the null pointer and crashes. The fix tests if the next-hop
array within a route is null or not. If the next-hop array is a null pointer
then ops-zebra does not process that route.

Change-Id: Ie0e39d7b81fa78803885435a3dae74add0bdd62f
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

Commit:
Code cleanup for compilation errors

When compiling with GCC 5, there are mores strict checks by the compiler
that revealed several flaws in the current code. Adressing the issues
raised by the compiler:
- Implicit functions declarations
- Missing or incorrect return values
- Invalid castings

Change-Id: I9b20945b90ce8a45633d597407fd7d7c9fc4e7cd
Signed-off-by: Diego Dompe <dompe@hpe.com>

Nimisha-Mathews committed:
chg: dev: IDL lookup optimization for BGP

This optimization involves caching and using UUIDs to avoid
expensive IDL searches while performing lookup operations, during
a route update or delete.

Change-Id: Ic7bdb7640d06a7f478df2ce4e579af47e2d46ada
Signed-off-by: Nimisha Sara Mathews <nimisha.sara.mathews@hpe.com>

buimy committed:
Support diag-dump on bgp

Change-Id: Idc9bab0d4120cb23af59d720e155d4a77bceccbb
Signed-off-by: buimy <myhanh.bui@hpe.com>

aslammdkhan committed:
chg: dev: Fix the Makefile error which was introduced while adding supportability for zebra static route, port and nexthop

TG-1008

Change-Id: Ibc36b478b647bd7fca85a03edf578c5250768f6d
Signed-off-by: aslammdkhan <aslam.khan@hpe.com>

gauravgupta4 committed:
new: dev: Tagia Bug#1110 Enhance zebra restart-ability test script to handle interface address changes

Added interface address change test cases to ops-zebra restart-ability
test script. The following two test cases need to be added:-

- While ops-zebra is down, change the interface address. All route
  next-hops should be cleaned from kernel once zebra comes back up.
  The selected bit on the roue and next-hops should be set accordingly.

- While ops-zebra is down, restore the interface address.
  All route next-hops should be re-programmed in kernel once zebra comes
  back up. The selected bit on the roue and next-hops should be set
  accordingly.

Change-Id: I572cde918722f2a715f81a0361a96774317fec4f
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

hpathpe committed:
Remove communication between BGP and Zebra deamon for the router_id update.

TG-714
Severity : High
Description: when the router ID is not configured in BGP router
configuration, one of the interface IP should be used as a Router ID.
This commit will remove the BGP router_id message passing communication
with Zebra. Zebra is writing active router id into the VRF table's
active_router_id column and BGPd is using this for setting BGP router's
router_id.

Change-Id: Ie53ff223d9e05795622f67e23f236ee2071c485e
Signed-off-by: Hitesh Padekar <hitesh.mar.padekar@hpe.com>

Nimisha-Mathews committed:
fix: usr: Routes not inserted into Route table
TG-1071

This fix addresses the issue that when large number of
routes are announced, all routes do not get inserted into
the Route table.

Change-Id: I0ab2ba58b7ca28191ebfeea22b751cc77c5e11f1
Signed-off-by: Nimisha Sara Mathews <nimisha.sara.mathews@hpe.com>

buzanogo committed:
Adding BGP Quagga test cases

Change-Id: I7d887befa1a01dd47d135c7ba9b3b86448c48aa8
Signed-off-by: Javier Buzano <buzano@hpe.com>

buzanogo committed:
Adding Zebra Quagga test cases

Change-Id: I14928bb7f4983bd0d2a4df5f3e73f8d45f1c8988
Signed-off-by: Javier Buzano <buzano@hpe.com>

buimy committed:
Refactor diag-dump on BGP to fix some bugs

Change-Id: Ice9d5a6475f0ead2613c350aefef253f6d02e3f3
Signed-off-by: buimy <myhanh.bui@hpe.com>

Commit:
Handle request for immediate run of OVSDB loop.

Taiga # 1150: Zebra stops processing Route (and other OVSDB tables)

The handling for the poll_loop timer was incorrect in the case where
an immediate re-run of the OVSDB loop is required. In this case, the
timeout_when value is LLONG_MIN (very negative). Handle this case
explicitly by looping instead of starting a timer.

Change-Id: I3715a0b0850d5130b46977ce7d82c15d8a4d06a7
Signed-off-by: DanCripe <dan@quattronetworks.com>

vijavaru committed:
fix : usr :
TG - 1168
Severity - High

Fix to enable routing for IPv6 networks advertised by BGP

Change-Id: I76c01b5275ae1661ab2d07295ae786797d25063b
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

lwang16 committed:
add bgp memory usage info to 'diag-dump bgp'

Tags - new, dev
TG - 239

Change-Id: Ifa3cad590d3766c03a202bef8a0319a76476b2ae
Signed-off-by: Lei Wang <lei.wang3@hpe.com>

Commit:
fix: dev: Fix for crash and abort related to ospfd.

Change-Id: I005f94e2b3b285d81eeebb36d208f70a3d227076
Signed-off-by: nisankas <satchit.nisanka@hpe.com>

gauravgupta4 committed:
new: dev: Taiga Bug#1259 ops-zebra batching route updates to OVSDB

-Added support for publishing batching updates to OVSDB
-Adding CTs for some scale tests within ops-zebra for
 IPv4 and IPv6 routes.

Change-Id: I5abd046db59007111b17381700cbbc9a7c467f04
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

vijavaru committed:
Revert "fix : usr : TG - 1168 Severity - High"

This reverts commit 113940800cdb5eda815dbbe769191992bc31fba0.

Change-Id: I830747914486ed646bec492615fc6be29857d500
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

fonsecamau committed:
chg: dev: Porting test cases to modular framework

Change-Id: I482b57e2f7b3a2a9a016e2119fd8cae5777e5635
Signed-off-by: fonsecamau <fonsecamau@hpe.com>

vijavaru committed:
fix : usr :
TG - 1168
Severity - High
Fix to enable routing for IPv6 networks advertised by BGP

Change-Id: Ia72dc8bb4c36e4985759518dcb39993a65c3c55b
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>
(cherry picked from commit 008b1f47e475c9a2acde406b581c0ba81d41c54a)

statapudi committed:
Zebra diagnostics support

TG-1304

-Add appctl command 'zebra/dump'
-Add support for 'diag-dump zebra basic' command
- Move diagnostics and debug functions to separate file

Change-Id: Ic844859e71dcc3dba3971702995ea8d97f5c8a6e
Signed-off-by: Surya Tatapudi <surya.tatapudi@hpe.com>
(cherry picked from commit e60b9584c0598186576ae6f5bd74d72ef263bb83)

vijavaru committed:
fix : dev:
TG - 1111
Severity - High

Since the bgp redistribute schema  has to be changed
( https://review.openswitch.net/#/c/10067/ ) to fix TG -1111
,the redistribute ops-quagga support has to be temporarily removed
to avoid build failures.
Once the new schema is merged, a new patch will be uploaded
to enable bgp redistribute ops-quagga support.

Change-Id: I026208ceb4d8c125000cdf3664b1e39e84c0f6c3
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>
(cherry picked from commit 8204cb6a6dd76a01b4aab87b5db1f5e448a8ee90)

ganeshragnarayanan committed:
fix BGP prefix list name corruption


While testing with multiple prefix-list, after certain iterations,
the name of the prefix-list data structure in plist.h gets corrupted.
This review fixes the issue

Tags: chg, usr

TG-1324

Change-Id: I87c74f3bb31d16c99e1e48247af86c7714f96777
Signed-off-by: Ganesh Narayanan <ganesh.rag.narayanan@hpe.com>

vijavaru committed:
fix : dev :
TG - 1108
Severity - High

ops-quagga changes for the schema changes in review https://review.openswitch.net/#/c/10067/

Change-Id: Iaf25c2befb0e5701a07712c0142fc20a31ef335f
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>
(cherry picked from commit 3456513a7148d18d8ca8006eec014d231b833221)

rohanatulshah committed:
Modularize zebra component

Moving zebra specific CLIs and tests to ops-quagga/zebra repository

TG-1034
Tags: new, usr

Change-Id: I91cdf89f507bde06f4d4a05c94c654de6e88aebb
Signed-off-by: Rohan Shah <rohan.shah@hpe.com>

sirigadi committed:
Fix segmentation fault due to NULL pointer dereference

TG-1331
Severity: High
Tags: fix, usr

Change-Id: I954ff5f6cca737994c4ec05a147bd4902c15ecaa
Signed-off-by: sirigadi <maheshwari.sirigadi@hpe.com>

hpathpe committed:
Updating CT for avoiding interfaces with overlapping networks.

TG-714

Change-Id: Ic2249807802db2fb698fd4c09a8304f46713aed8
Signed-off-by: Hitesh Padekar <hitesh.mar.padekar@hpe.com>

gauravgupta4 committed:
new: dev: Tagia Bug#1110 Make sleep time uniform zebra component test cases

- Add uniform sleep time for zebra modular component test cases

Change-Id: Ie8954375932ba326891131585d4e405e1ac06ac3
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

rohanatulshah committed:
Enabling skipped test cases post zebra cli modularization

Tags: chg, dev
TG-1034

Change-Id: I1b824e790ba36537c1220dcabf264c9b8f6b2fd5
Signed-off-by: Rohan Shah <rohan.shah@hpe.com>

gauravgupta4 committed:
Move connected routes ownership to ops-zebra from ops-portd

- Added support for maintenance of connected routes into ops-zebra
- ops-zebra adds, deletes connected routes for interface addresses
  into OVSDB
- ops-zebra set and un-sets the selected bits for connected route
  and its next-hop, if the interface in "no shut/shut" respectivly
- ops-zebra supports clean-up of connected routes from OVSDB if the
  are turned into L2 using "no routing" or the interfaces are deleted
  using CLI.
- ops-zebra also handles the clean-up and update of connected routes
  after zebra comes up from a restart.
- The component test script uses different interface types and
  aforementioned triggers to test this enhancement. This is being
  skipped as of now.

TG-1308
Tags: fix, usr

Change-Id: Iaecec351875559fa4f97dae50997622e107e888d
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

hpathpe committed:
BGP unused TCP port 2605 must be closed.

TG-1443
Tags: fix, usr

Change-Id: I4dd2404fbabc791b6eaf58762771178755419510
Signed-off-by: Hitesh Padekar <hitesh.mar.padekar@hpe.com>

rohanatulshah committed:
Close the unused TCP port 2601 used by OpenSource zebra protocol

Change-Id: I6b8bbf732fb04d8d83f960024d7aeaaa33b21f97
Signed-off-by: Rohan Shah <rohan.shah@hpe.com>

gauravgupta4 committed:
Enabling connected routes CT in ops-zebra

TG-1308
Tags: fix, usr

Change-Id: I5fe8e2af68e7ff05b588d1b7c4da994b0caadd12
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

memon4hp committed:
close the unused TCP port 2604 by ops-ospfd

TG-1468
Tags: chg, dev

Change-Id: Ic4aa8bdc3425eff62899f7dba045d2e0a545fa57
Signed-off-by: Mahammadnaeem Memon <memon.mk@hpe.com>

gauravgupta4 committed:
Skipping test_zebra_ct_ecmp.py due to failures in ops-quagga gate

TG-1308
Tags: fix, usr

Change-Id: I1019b7a7b1b45497f4b353d6b9a7a16ed749dc7e
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

gauravgupta4 committed:
ops-zebra selecting routes in OVSDB erronously

This fixes the following two issues:-
- ops-zebra selecting routes in OVSDB without causing the next-hop
  selection criteria to run. This is causing the backup hops to get
  selected in OVSDB in case of interface state change triggers.
- In case of "no routing/interface delete" triggers, ops-zebra would
  delete BGP and OSPF protocol next-hops from DB. ops-zebra just needs
  to process the next-hops from static routes.
- In case of deletion of L3 interface using "no interface loopback 1",
  zebra does not cleanup the static routes routing via that interface.
  ops-zebra should trigger a clean-up such next-hops.

TG-1440
Tags: fix, usr

Change-Id: I90e1ac2b2fc75f9ce376e5c7dd937b077833fe31
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

Test214-dev committed:
Impossible to set zero BGP keepalive/holdtime

Tags: fix, dev
TG-1517

Change-Id: Ibc8e9d5e84c1e4b6d093a5067156a5b2fdf790ff
Signed-off-by: Ilya Schepin <schepin.io@gmail.com>

gauravgupta4 committed:
ops-zebra not cleaning up OVSDB in case of "no routing"

THis fix addresses a corner case of "no routing" when the L3 interface
does not have an IP/IPv6 address but this interface is L3 since it is the
part of vrf. In case of "no routing" in this kind of interface, zebra should
clean-up the DB for the route going via this interface.

TG-1502
Tags: fix, usr

Change-Id: I421819be7a85f10b38d497da7cadf6a76e8373ac
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

matthew-m-hall committed:
new: dev: add layer3 tests

TG-1332

Change-Id: I58857d19794dd3d1666cb797f8ca330292b957f6
Signed-off-by: Matthew Hall <matthew.hall@hpe.com>

rohanatulshah committed:
Spare the zera zserv socket as it is used by BGP for internal communications

TG-1441
Tags: chg, dev

Change-Id: Iaba56cf4ce6184059efdb73ad24fe3a291379bdf
Signed-off-by: Rohan Shah <rohan.shah@hpe.com>

gauravgupta4 committed:
Kernel routes are missing in case of protocol ECMP

This defect addresses the following two issues:-
- ops-zebra does not add ECMP protocol (BGP, OSPF) routes
  into kernel
- ops-zebra currently deletes all protocol (BGP, OSPF)
  next-hops for a route even if only next-hop is deleted

TG-1552
Tags: fix, usr

Change-Id: I3aeb8363fc31da0626aa9c47d9f10932df54cf76
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

gauravgupta4 committed:
Conversion of ops-zebra restart-ability CT script to modular framework

TG-1110
Tags: new, dev

Change-Id: Ie86b8f27ba40fbc955b25775bb92b82a8f395013
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

memon4hp committed:
close the unused TCP port 2604 by ops-ospfd cleanup

TG-1468
Tags: chg, dev

Change-Id: I6096670c0716f9cdd35ef7b7eab664c5700455c0
Signed-off-by: Mahammadnaeem Memon <memon.mk@hpe.com>

vijavaru committed:
fix, dev: Fix to display BGP neighor reset reason

TG - 1602
Fix to display BGP neighbor reset reason in “diag-dump bgp basic” CLI
output
Change-Id: I06887bfb8846357812d3379d66793d40f8de3a8d
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>

gauravgupta4 committed:
Static routes not removed from OVSDB when related interface is set to "no routing" in a scaled setup

This fix addresses the following issues:-
- In a scaled setup, ops-zebra won't remove all static routes if the resolving
  L3 port is made L2 by "no routing". This is because ops-zebra doesn't create
  a new transaction once it has published updates for 100 routes to OVSDB.
- ops-zebra will crash in case a kernel route gets deleted due to NULL pointer
  de-refrence.
- Enabling ops-zebra scaled IPV4 and IPV6 route test cases.
- Fixed the name of the loopback interfaces in the zebra CTs. The loopback
  interface name has been updated to 'loopback1' instead of 'lo1' previously
- Adding the BGP fix at https://review.openswitch.net/#/c/13556/ for getting
  all quagga CT issues resolved via one commit.
- Committing the fix for OSPFv2 test script failure in the FT file
  ops-tests/feature/ospf/test_ospfv2_ft_best_route.py

TG-1601
Tags: fix, usr

Change-Id: I246a8e3c19b61a2f144e6b00de160ab66f528845
Signed-off-by: Gaurav Gupta <gaurav.gupta4@hpe.com>

vijavaru committed:
chg, usr: Changing BGP to eBGP in the Quagga Design documentation

TG - 1589
Changing BGP to eBGP. This terminology will need to
change on OpenSwitch.net and in the Toronto documentation.
Change-Id: I8cc5460ad2c089668ae7e24f4d451e0e1095ac15
Signed-off-by: vijavaru <varun.vijayakumar@hpe.com>
(cherry picked from commit fd99ec495b0b262dbdbdaa66491ae30cf63f2ab1)

Nimisha-Mathews committed:
5712 BGP connection establishment delay on reboot
TG-1818

Tags: fix, usr

Change-Id: Ic3fbf2fb5680bd33aab933c950bf5592fcda9607
Signed-off-by: Nimisha Sara Mathews <nimisha.sara.mathews@hpe.com>
(cherry picked from commit 1ee4a5b4828d89c384ece20a93ab84449ccc365e)

stroykov commented on zebra/cli/l3routes_vty.c:
```diff
@@ -1751,7 +1719,7 @@ int route_prefix_comparator(const void *route_entry1,
     if (ip1 && ip2) {
         if (is_ipv6_prefix1_valid && is_ipv6_prefix2_valid) {
-            strcmp_result = ipv6_prefix_compare(&ipv6_str1, &ipv6_str2);
+            strcmp_result = IPV6_ADDR_CMP(&ipv6_str1, &ipv6_str2);
```
input must be binary

xschily replied:
It is a binary, filled with inet_pton on line 1711 and 1715. Don't know why it called ipv6_str.

stroykov replied:
you have a chance to make the code more readable:)

stroykov commented on bgpd/bgp_ovsdb_route.c:
```diff
@@ -417,7 +417,7 @@ bgp_ovsdb_lookup_local_nexthop(char *ip)
         if (row->ip_address)
         {
-            if (strcmp(ip, row->ip_address) == 0) {
+            if (ip_addr_cmp(ip, row->ip_address)) {
```
should we expect the same result/behavior  as with the strcmp?

xschily replied:
I think it's better when function returns positive value for positive result.

stroykov replied:
positive or non zero?
if you want to use bool result, than you should use something like "is_equal". If you extend "cmp" functions family, you should use cmp semantic

stroykov commented on zebra/cli/l3routes_vty.c:
```diff
@@ -136,7 +136,7 @@ port_find_ipaddress (const struct ovsrec_port *port_row, char * ip_entry,
                       sizeof(ip6_addrcopy));
               ipv6_address = strtok(ip6_addrcopy, "/");
-              if (!strcmp (ipv6_address, ip_entry))
+              if (ip_addr_cmp_af (AF_INET6, ipv6_address, ip_entry))
```
I guess we should use the same function ip_addr_cmp everywhere

xschily replied:
Well in case if we already know exactly which AF we use ip_addr_cmp_af is better. ip_addr_cmp is used to compare addresses when AF is unknown to the caller. Still, I can replace all calls to ip_addr_cmp_af with calls to ip_addr_cmp, but it will be a bit slower, though easier to understand. Or I can merge 2 functions into 1, like  

```
ip_addr_cmp (int af, char *addr_str1, char *addr_str2) {
    switch (af) {
        case AF_INET: ...
        case AF_INET6: ....
        default: af = ((strstr(addr_str1, ":") != NULL) ? AF_INET6 : AF_INET)
```

vkarpov commented:
No comments.

xschily committed:
Full notation IPv6 addresses

In Quagga function strcmp is used to compare both ipv4 and ipv6 addresses.
This is incorrect, since there are multiple string representations of one
ipv6 address (e.g. strings 2002::a1, 2002:0:0:0:0:0:0:a1, 2002::00a1,
2002::00A1 represent one ipv6 address, 2002:0000:0000:0000:0000:0000:0000:00a1.
Possible representations are described in RFC 5952). To correctly compare
ipv6 addresses, they must be compared using IPV6_ADDR_CMP function (defined
in lib/prefix.h), which will compare addresses stored in struct in6_addr form.

In some cases calling function is not aware of underlying string address
family (AF_INET or AF_INET6). So wrapping function ip_addr_is_equal is created,
to determine AF of IP address string.

Tags: fix, dev, chg

Change-Id: I9a11038b0ac50613601d1ca56ce97388fbb84dbb
Signed-off-by: Ilya Schepin <ischepin@mera.ru>

xschily committed:
Full notation IPv6 addresses

Tags: fix, dev

Change-Id: I9a11038b0ac50613601d1ca56ce97388fbb84dbb
Signed-off-by: Ilya Schepin <ischepin@mera.ru>

xschily committed:
Full notation IPv6 addresses

Tags: fix, dev, Github issue #7

Change-Id: I9a11038b0ac50613601d1ca56ce97388fbb84dbb
Signed-off-by: Ilya Schepin <ischepin@mera.ru>

stroykov commented on lib/prefix.c:
```diff
@@ -903,4 +903,25 @@ inet6_ntoa (struct in6_addr addr)
   inet_ntop (AF_INET6, &addr, buf, INET6_ADDRSTRLEN);
   return buf;
 }
+
```
where is result type defenition?

stroykov committed:
Merge branch 'full_notation_ipv6_addr' of https://github.com/MERAprojects/ops-quagga into full_notation_ipv6_addr

Conflicts:
	lib/prefix.c

Change-Id: Id6b0d8e9e887aa055a66f2f516c8ea8d4fef7ba1

apolovin reviewed (approved):
OK for me

stroykov commented:
https://review.openswitch.net/#/c/13886/

stroykov committed:
Merge branch 'full_notation_ipv6_addr' of https://github.com/MERAprojects/ops-quagga into full_notation_ipv6_addr
